ocaml/xapi/xapi_clustering.ml

```diff
@@ -423,133 +423,138 @@ let compute_corosync_max_host_failures ~__context =
   in
   corosync_ha_max_hosts
 
-let on_corosync_update ~__context ~cluster updates =
-  debug
-    "%s: Received %d updates from corosync_notifyd , run diagnostics to get \
-     new state"
-    __FUNCTION__ (List.length updates) ;
-  let m =
-    Cluster_client.LocalClient.diagnostics (rpc ~__context)
-      "update quorum api fields with diagnostics"
-  in
-  match Idl.IdM.run @@ Cluster_client.IDL.T.get m with
-  | Ok diag ->
-      Db.Cluster.set_is_quorate ~__context ~self:cluster ~value:diag.is_quorate ;
-      let all_cluster_hosts = Db.Cluster_host.get_all ~__context in
-      let ip_ch =
-        List.map
-          (fun ch ->
-            let pIF = Db.Cluster_host.get_PIF ~__context ~self:ch in
-            let ipstr =
-              ip_of_pif (pIF, Db.PIF.get_record ~__context ~self:pIF)
-              |> ipstr_of_address
-            in
-            (ipstr, ch)
-          )
-          all_cluster_hosts
-      in
-      let current_time = API.Date.now () in
-      ( match diag.quorum_members with
-      | None ->
-          List.iter
-            (fun self ->
-              Db.Cluster_host.set_live ~__context ~self ~value:false ;
-              Db.Cluster_host.set_last_update_live ~__context ~self
-                ~value:current_time
+module Watcher = struct
+  let on_corosync_update ~__context ~cluster updates =
+    debug
+      "%s: Received %d updates from corosync_notifyd , run diagnostics to get \
+       new state"
+      __FUNCTION__ (List.length updates) ;
+    let m =
+      Cluster_client.LocalClient.diagnostics (rpc ~__context)
+        "update quorum api fields with diagnostics"
+    in
+    match Idl.IdM.run @@ Cluster_client.IDL.T.get m with
+    | Ok diag ->
+        Db.Cluster.set_is_quorate ~__context ~self:cluster
+          ~value:diag.is_quorate ;
+        let all_cluster_hosts = Db.Cluster_host.get_all ~__context in
+        let ip_ch =
+          List.map
+            (fun ch ->
+              let pIF = Db.Cluster_host.get_PIF ~__context ~self:ch in
+              let ipstr =
+                ip_of_pif (pIF, Db.PIF.get_record ~__context ~self:pIF)
+                |> ipstr_of_address
+              in
+              (ipstr, ch)
             )
             all_cluster_hosts
-      | Some nodel ->
```

Comment:
git diff feels kind of strange here, but I don't I have copied any function wrong...